account/account.go

```diff
@@ -293,6 +295,27 @@ func (account *Account) PrecomputeAddress(deployerAddress *felt.Felt, salt *felt
 
 }
 
+// WaitForTransactionReceipt waits for the transaction to succeed or fail
+func (account *Account) WaitForTransactionReceipt(ctx context.Context, transactionHash *felt.Felt, pollInterval time.Duration) (*rpc.TransactionReceipt, error) {
+	t := time.NewTicker(pollInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, ctx.Err()
+		case <-t.C:
+			receipt, err := account.TransactionReceipt(ctx, transactionHash)
+			if err != nil {
+				if err.Error() == rpc.ErrHashNotFound.Error() {
+					continue
```

Comment:
Is it ok that in case of wrong tx hash it's going to poll over and over again?

Reply:
I was wondering the same thing. I think it should loop until the context cancels. I was tempted to add an additional paramter to specify the timeout, but in principle the context should do this (afaik), so I decided to leave it as is.

Reply:
I don't think there's a distinction between incorrect tx hash and a tx hash that hasn't been added to the chain. This is why users should specify a timeout in the context.